packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/Tasks.spec.js

```diff
@@ -1,133 +1,188 @@
-  const classification = classifications.active
-  const { addAnnotation } = classifications
-  singleChoiceTask.setAnnotation(annotation)
-
-  it('should render without crashing', function () {
-    const wrapper = shallow(<Tasks />)
-    expect(wrapper).to.be.ok()
-  })
+  let addAnnotation
+  let classification
+  let step
+  let tasks
+  let TaskComponent
 
-  it('should render null on initialization', function () {
-    const wrapper = shallow(<Tasks />)
-    expect(wrapper.type()).to.be.null()
-  })
+  const taskTypes = Object.keys(taskRegistry.register)
 
-  it('should render a loading UI when the workflow loading', function () {
-    const wrapper = shallow(<Tasks loadingState={asyncStates.loading} />)
-    expect(wrapper.contains('Loading')).to.be.true()
+  before(function () {
+    // Mute the onAction warnings for the duration of these tests.
+    sinon.stub(console, 'error')
   })
 
-  it('should render an error message when there is a loading error', function () {
-    const wrapper = shallow(<Tasks loadingState={asyncStates.error} />)
-    expect(wrapper.contains('Something went wrong')).to.be.true()
+  after(function () {
+    console.error.restore()
   })
 
-  it('should render null if the workflow is load but has no tasks', function () {
-    const wrapper = shallow(<Tasks loadingState={asyncStates.success} ready />)
-    expect(wrapper.type()).to.be.null()
-  })
+  taskTypes.forEach(function (taskType) {
+    before(function () {
+      const task = taskRegistry.get(taskType)
+      TaskComponent = observer(task.TaskComponent)
+      const taskSnapshot = {
+        instruction: `${taskType} instructions`,
+        question: `${taskType} question`,
+        taskKey: 'init',
+        type: taskType
+      }
+      const workflowSnapshot = WorkflowFactory.build({
+        id: 'tasksWorkflow',
+        display_name: 'A test workflow',
+        tasks: {
+          init: taskSnapshot
+        },
+        version: '0.0'
+      })
+      const subjectSnapshot = SubjectFactory.build({
+        id: 'subject',
+        metadata: {}
+      })
+      const projectSnapshot = ProjectFactory.build({
+        id: 'project'
+      })
+      const rootStore = RootStore.create({
+        projects: {
+          active: projectSnapshot.id,
+          resources: {
+            [projectSnapshot.id]: projectSnapshot
+          }
+        },
+        subjects: {
+          active: subjectSnapshot.id,
+          resources: {
+            [subjectSnapshot.id]: subjectSnapshot
+          }
+        },
+        workflows: {
+          active: workflowSnapshot.id,
+          resources: {
+            [workflowSnapshot.id]: workflowSnapshot
+          }
+        }
+      }, {
+        authClient: {
+          checkBearerToken: sinon.stub().callsFake(() => Promise.resolve(null)),
+          checkCurrent: sinon.stub().callsFake(() => Promise.resolve(null))
+        },
+        client: {
+          panoptes: {
+            get: sinon.stub().callsFake(() =>
```

Comment:
Good point. It should work here. I’ve been using it in other PRs, and I’ve found one bug that I should write up an issue for. There are circumstances where it will return `[ undefined ]` when it should return `[]`.